spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/ModuleDefinitionBuilder.java

```diff
@@ -87,29 +89,36 @@ else if (argument.getName().equalsIgnoreCase("outputType")) {
 				}
 			}
 			if (m > 0) {
-				builder.setParameter(BindingProperties.INPUT_BINDING_KEY, String.format("%s.%d", streamName, m - 1));
-				builder.setParameter(BindingProperties.INPUT_GROUP_KEY, DEFAULT_CONSUMER_GROUP_NAME);
+				builder.setParameter(BindingProperties.INPUT_BINDING_KEY,
+						String.format("%s.%s", streamName, moduleNodes.get(m - 1).getLabelName()));
+				updateConsumerGroupName(builder, DEFAULT_CONSUMER_GROUP_NAME);
 				builder.setParameter(BindingProperties.INPUT_DURABLE_SUBSCRIPTION_KEY, "true");
 			}
 			if (m < moduleNodes.size() - 1) {
 				builder.setParameter(BindingProperties.OUTPUT_BINDING_KEY,
-						String.format("%s.%d", streamName, m));
+						String.format("%s.%s", streamName, moduleNode.getLabelName()));
 			}
 			builders.add(builder);
 		}
 		SourceChannelNode sourceChannel = streamNode.getSourceChannelNode();
 		if (sourceChannel != null) {
-			String sourceChannelName = removeChannelTypePrefix(sourceChannel.getChannelName());
-			builders.getLast().setParameter(BindingProperties.INPUT_BINDING_KEY,
-					sourceChannelName);
-			builders.getLast().setParameter(BindingProperties.INPUT_GROUP_KEY, streamName);
-			builders.getLast().setParameter(BindingProperties.INPUT_DURABLE_SUBSCRIPTION_KEY, "true");
+			ModuleDefinition.Builder sourceModuleBuilder = builders.getLast();
+			sourceModuleBuilder.setParameter(BindingProperties.INPUT_BINDING_KEY, sourceChannel.getChannelName());
+			String consumerGroupName = streamName;
+			if (sourceChannel.getArguments() != null) {
```

Comment:
I'm with supporting declaring some specific user group on the DSL. I guess it won't be done very frequently, but I suppose it's good to have it there, since it's a first class feature of the consumers. By default it would be set to the stream name as far as my reading of the code goes, correct? That's what we want, because different streams get different consumer groups then by default.